docs(tutorial): lamp ubuntu 22.04 (#582)
scaleway/docs-content: tutorials/setup-lamp-stack-ubuntu2204/index.mdx

tags: LAMP Ubuntu PHP Apache MySQL Linux
categories:
  - compute
  - instances
dates:
  validation: 2022-04-28
  posted: 2022-04-28
---

## LAMP stack overview

The LAMP stack references a group of software that can be used to serve dynamic web applications and websites. LAMP is an acronym and stands for **L**inux, **A**pache, **M**ySQL, and **P**HP). 
The concept describes a server environment that consists of a Linux operating system (Ubuntu Jammy Jellyfish 22.04 LTS), an Apache web server, a MySQL (or MariaDB) database server, and the PHP programming language. 

<Message type="requirement">

  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have [configured your SSH key](/console/my-project/how-to/create-ssh-key/)
  - You have a [Scaleway Instance](/compute/instances/how-to/create-an-instance/) running on Ubuntu 22.04 (Jammy Jellyfish)
  - You have a FQDN (Fully Qualified Domain Name) pointing to your Instance’s IP address
  - You have sudo privileges or access to the root user

</Message>

## Installing the LAMP stack on an Ubuntu 22.04 Instance

> 1. [Connect to your Instance via SSH](/compute/instances/how-to/connect-to-instance/)

Reviewer: ```suggestion
1. [Connect to your Instance via SSH](/compute/instances/how-to/connect-to-instance/).
```

2. Use the `apt` package manager of Ubuntu to install the required packages. Make sure that the system is up to date and has the latest bug-fixes and updates installed by running the following command:

  ```bash
  apt update && apt -y upgrade
  ```

3. Install the software stack:

  ```bash
  apt install -y ufw apache2 mariadb-server php-fpm php-mysql
  ```

## Securing the database engine

1. Launch the configuration wizard to configure the database server:

  ```bash
  mysql_secure_installation
  ```

2. Press **Enter** when prompted for the current root password for the MariaDB server, as the password is not yet set.

3. Press `Y` to switch to `unix_socket` authentication:  
    ```
    Switch to unix_socket authentication [Y/n]
    ```

4. Press `N` to keep your current root password, press `Y` to change the root password:

  ```bash
  You already have your root account protected, so you can safely answer 'n'.

  Change the root password? [Y/n]
  ```

> 5. If you decide to change the root password, enter the new password and press `Enter`, then repeat this step.

Reviewer 2: ```suggestion
5. Enter the new password if you decided to change the root password, press `Enter`, then repeat this step.
```

6. Keep the settings proposed by default when prompted by further questions, by pressing `Enter` each time.

## Configuring an Apache virtual host

> Nginx stores the configuration of virtual hosts configuration files. All available configuration files are located in the `/etc/apache2/sites-available/` directory.

Reviewer: ```suggestion
Apache stores the configuration of virtual hosts configuration files. All available configuration files are located in the `/etc/apache2/sites-available/` directory.
```

5. Enable the `proxy_fcgi` module:
    ```
    a2enmod proxy_fcgi
    ```

6. Reload the Apache configuration to make the site available. 

  ```bash
  systemctl reload apache2
  ```

## Securing your Instance with a firewall

1. Enable HTTP, HTTPS and SSH connections in the [firewall](/tutorials/installation-uncomplicated-firewall/) configuration of the Instance by running the following command:

  ```bash
  ufw allow 'Apache Full'
  ufw allow 'OpenSSH'
  ```

2. Enable ufw:

  ```bash
  ufw enable
  ```

  Confirm the activation of the firewall by typing `y`:

  ```bash
  Command may disrupt existing ssh connections. Proceed with operation (y|n)? y
  Firewall is active and enabled on system startup
  ```

3. Check the status of ufw:

  ```bash
  ufw status
  ```
  
  It will return a list of the allowed services:

   ```bash
    Status: active

    To                         Action      From
    --                         ------      ----
    Apache Full                ALLOW       Anywhere
    OpenSSH                    ALLOW       Anywhere
    Apache Full (v6)           ALLOW       Anywhere (v6)
    OpenSSH (v6)               ALLOW       Anywhere (v6)
  ```

## Testing PHP

> 1. Create a test PHP file to test if your LEMP stack is working.

Reviewer: ```suggestion
1. Create a test PHP file to test if your LAMP stack is working.
```

    Successfully received certificate.
    Certificate is saved at: /etc/letsencrypt/live/example.com/fullchain.pem
    Key is saved at:         /etc/letsencrypt/live/example.com/privkey.pem
    This certificate expires on 2022-07-27.
    These files will be updated when the certificate renews.
    Certbot has set up a scheduled task to automatically renew this certificate in the background.

    Deploying certificate
    Successfully deployed certificate for example.com to /etc/apache2/sites-available/example.com-le-ssl.conf
    Successfully deployed certificate for www.example.com to /etc/apache2/sites-available/example.com-le-ssl.conf
    Congratulations! You have successfully enabled HTTPS on https://example.com and https://www.example.com

    - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    If you like Certbot, please consider supporting our work by:
    * Donating to ISRG / Let's Encrypt:   https://letsencrypt.org/donate
    * Donating to EFF:                    https://eff.org/donate-le
    - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    ```

> 5. Reload the Nginx configuration.

Reviewer: ```suggestion
5. Reload the Apache configuration.
```
